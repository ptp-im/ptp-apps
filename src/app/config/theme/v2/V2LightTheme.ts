import color from 'color';
import { MD2LightTheme } from 'react-native-paper';
import type { MD2Theme } from 'react-native-paper';

import { black, red800, white } from './colors';

export const V2LightTheme: MD2Theme = {
  ...MD2LightTheme,
  dark: false,
  roundness: 4,
  version: 2,
  isV3: false,
  colors: {
    ...MD2LightTheme.colors,
    // primary: '#6200ee',
    accent: '#03dac4',
    // background: '#f6f6f6',
    // background: grey50,
    surface: white,
    error: '#B00020',
    // text: black,
    // onSurface: '#000000',
    disabled: color(black).alpha(0.26).rgb().string(),
    placeholder: color(black).alpha(0.54).rgb().string(),
    backdrop: color(black).alpha(0.5).rgb().string(),
    // notification: pinkA400,

    msgUnreadBadge: red800,
    onSurface: 'rgb(30, 107, 192)',
    primary: 'rgb(30, 107, 192)',
    background: 'rgb(235, 235, 240)',
    white: 'rgb(255, 255, 255)',
    background1: 'rgb(235, 235, 240)',
    backgroundSelected: 'rgb(62, 125, 185)',
    backgroundTabs: 'rgb(255, 255, 255)',
    card: 'rgb(255, 255, 255)',
    text: 'rgb(28, 28, 30)',
    textPrimary: 'rgb(10, 132, 255)',
    textAlert: 'rgb(255, 59, 48)',
    border: 'rgb(229, 229, 229)',
    notification: 'rgb(255, 59, 48)',
  },
};

export const V2NavLightColors = {
  primary: V2LightTheme.colors.primary,
  background: V2LightTheme.colors.background,
  card: V2LightTheme.colors.card,
  text: V2LightTheme.colors.text,
  border: V2LightTheme.colors.border,
  notification: V2LightTheme.colors.notification,
};
