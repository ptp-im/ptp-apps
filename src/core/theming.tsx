import type { ColorValue } from 'react-native';

import { createTheming } from '@callstack/react-theme-provider';
import {
  argbFromHex,
  themeFromSourceColor,
} from '@material/material-color-utilities';
import color from 'color';

import {
  MD2DarkTheme,
  MD2LightTheme,
  MD3DarkTheme,
  MD3LightTheme,
} from '../styles/themes';
import { tokens } from '../styles/themes/v3/tokens';
import type {
  Theme,
  MD2Theme,
  MD3Theme,
  MD3Colors,
  MD3AndroidColors,
} from '../types';

export const DefaultTheme = MD3LightTheme;

const {
  ThemeProvider,
  withTheme,
  useTheme: useThemeProviderTheme,
} = createTheming<Theme>(DefaultTheme);

const useTheme = (overrides?: Parameters<typeof useThemeProviderTheme>[0]) =>
  useThemeProviderTheme<MD2Theme | MD3Theme>(overrides);

export { ThemeProvider, withTheme, useTheme };

export const defaultThemesByVersion = {
  2: {
    light: MD2LightTheme,
    dark: MD2DarkTheme,
  },
  3: {
    light: MD3LightTheme,
    dark: MD3DarkTheme,
  },
};

export const getTheme = (isDark = false, isV3 = true) => {
  const themeVersion = isV3 ? 3 : 2;
  const scheme = isDark ? 'dark' : 'light';

  return defaultThemesByVersion[themeVersion][scheme];
};

type Config = {
  sourceColor: ColorValue;
};

type Schemes = {
  lightScheme: MD3Colors;
  darkScheme: MD3Colors;
};

export const getDynamicThemeElevations = (scheme: MD3AndroidColors) => {
  const elevationValues = ['transparent', 0.05, 0.08, 0.11, 0.12, 0.14];
  return elevationValues.reduce((elevations, elevationValue, index) => {
    return {
      ...elevations,
      [`level${index}`]:
        index === 0
          ? elevationValue
          : color(scheme.surface)
              .mix(color(scheme.primary), elevationValue as number)
              .rgb()
              .string(),
    };
  }, {});
};

export const createDynamicThemeColors = ({ sourceColor }: Config): Schemes => {
  const { opacity } = tokens.md.ref;
  const modes = ['light', 'dark'] as const;

  const { schemes, palettes } = themeFromSourceColor(
    argbFromHex(color(sourceColor).hex())
  );

  const { light, dark } = modes.reduce(
    (prev, curr) => {
      const schemeModeJSON = schemes[curr].toJSON();

      const customColors = {
        surfaceDisabled: color(schemeModeJSON.onSurface)
          .alpha(opacity.level2)
          .rgb()
          .string(),
        onSurfaceDisabled: color(schemeModeJSON.onSurface)
          .alpha(opacity.level4)
          .rgb()
          .string(),
        backdrop: color(palettes.neutralVariant.tone(20))
          .alpha(0.4)
          .rgb()
          .string(),
      };

      const elevation = getDynamicThemeElevations(schemeModeJSON);

      const dynamicThemeColors = Object.assign(
        {},
        ...Object.entries(schemeModeJSON).map(([colorName, colorValue]) => ({
          [colorName]: color(colorValue).rgb().string(),
        })),
        {
          elevation,
          ...customColors,
        }
      );

      return {
        ...prev,
        [curr]: dynamicThemeColors,
      };
    },
    { light: MD3LightTheme.colors, dark: MD3DarkTheme.colors }
  );

  return { lightScheme: light, darkScheme: dark };
};
