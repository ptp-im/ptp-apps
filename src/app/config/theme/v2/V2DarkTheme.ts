import color from 'color';

import type { MD2Theme } from 'react-native-paper';
import { black, red800, white } from './colors';
import { V2LightTheme } from './V2LightTheme';

export const V2DarkTheme: MD2Theme = {
  ...V2LightTheme,
  dark: true,
  mode: 'adaptive',
  version: 2,
  isV3: false,
  colors: {
    ...V2LightTheme.colors,
    // primary: '#BB86FC',
    accent: '#03dac6',
    // background: '#121212',
    surface: '#121212',
    error: '#CF6679',
    onSurface: '#FFFFFF',
    // text: white,
    disabled: color(white).alpha(0.38).rgb().string(),
    placeholder: color(white).alpha(0.54).rgb().string(),
    backdrop: color(black).alpha(0.5).rgb().string(),
    // notification: pinkA100,

    msgUnreadBadge: red800,
    primary: 'rgb(10, 132, 255)',
    background: 'rgb(1, 1, 1)',
    card: 'rgb(18, 18, 18)',
    text: 'rgb(229, 229, 231)',
    textPrimary: 'rgb(10, 132, 255)',
    border: 'rgb(39, 39, 41)',
    notification: 'rgb(255, 69, 58)',
  },
};

export const V2NavDarkColors = {
  primary: V2LightTheme.colors.primary,
  background: V2LightTheme.colors.background,
  card: V2LightTheme.colors.card,
  text: V2LightTheme.colors.text,
  border: V2LightTheme.colors.border,
  notification: V2LightTheme.colors.notification,
};
