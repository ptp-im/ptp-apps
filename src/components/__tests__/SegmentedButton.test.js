import * as React from 'react';

import { render } from '@testing-library/react-native';
import color from 'color';
import renderer from 'react-test-renderer';

import { getTheme } from '../../core/theming';
import { black } from '../../styles/themes/v2/colors';
import SegmentedButtons from '../SegmentedButtons/SegmentedButtons';
import {
  getDisabledSegmentedButtonStyle,
  getSegmentedButtonColors,
} from '../SegmentedButtons/utils';

it('renders segmented button', () => {
  const tree = renderer
    .create(
      <SegmentedButtons
        onValueChange={() => {}}
        value={'walk'}
        buttons={[{ value: 'walk' }, { value: 'ride' }]}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders disabled segmented button', () => {
  const tree = renderer
    .create(
      <SegmentedButtons
        onValueChange={() => {}}
        value={'walk'}
        buttons={[{ value: 'walk' }, { value: 'ride', disabled: true }]}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders checked segmented button with selected check', () => {
  const tree = renderer
    .create(
      <SegmentedButtons
        onValueChange={() => {}}
        value={'walk'}
        buttons={[
          { value: 'walk', showSelectedCheck: true },
          { value: 'ride', disabled: true },
        ]}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

describe('getSegmentedButtonColors', () => {
  it('should return correct background color when checked and theme version 3', () => {
    expect(
      getSegmentedButtonColors({
        theme: getTheme(),
        disabled: false,
        checked: true,
      })
    ).toMatchObject({ backgroundColor: getTheme().colors.secondaryContainer });
  });

  it('should return correct background color when checked and theme version 2', () => {
    expect(
      getSegmentedButtonColors({
        theme: getTheme(false, false),
        disabled: false,
        checked: true,
      })
    ).toMatchObject({
      backgroundColor: color(getTheme(false, false).colors.primary)
        .alpha(0.12)
        .rgb()
        .string(),
    });
  });

  it('should return correct background color when uncheked (V3 & V2)', () => {
    expect(
      getSegmentedButtonColors({
        theme: getTheme(false, false),
        disabled: false,
        checked: false,
      })
    ).toMatchObject({
      backgroundColor: 'transparent',
    });
  });

  it('should return correct border color with theme version 3', () => {
    expect(
      getSegmentedButtonColors({
        theme: getTheme(),
        disabled: false,
        checked: false,
      })
    ).toMatchObject({
      borderColor: getTheme().colors.outline,
    });
  });

  it('should return correct border color with theme version 2', () => {
    expect(
      getSegmentedButtonColors({
        theme: getTheme(false, false),
        disabled: false,
        checked: false,
      })
    ).toMatchObject({
      borderColor: color(black).alpha(0.29).rgb().string(),
    });
  });

  it('should return correct border color when disabled and theme version 3', () => {
    expect(
      getSegmentedButtonColors({
        theme: getTheme(),
        disabled: true,
        checked: false,
      })
    ).toMatchObject({
      borderColor: getTheme().colors.surfaceDisabled,
    });
  });

  it('should return correct textColor with theme version 3', () => {
    expect(
      getSegmentedButtonColors({
        theme: getTheme(),
        disabled: false,
        checked: false,
      })
    ).toMatchObject({
      textColor: getTheme().colors.onSurface,
    });
  });

  it('should return correct textColor with theme version 2', () => {
    expect(
      getSegmentedButtonColors({
        theme: getTheme(false, false),
        disabled: false,
        checked: false,
      })
    ).toMatchObject({
      textColor: getTheme(false, false).colors.primary,
    });
  });

  it('should return correct textColor when disabled and theme version 3', () => {
    expect(
      getSegmentedButtonColors({
        theme: getTheme(),
        disabled: true,
        checked: false,
      })
    ).toMatchObject({
      textColor: getTheme().colors.onSurfaceDisabled,
    });
  });

  it('should return correct textColor when disabled and theme version 2', () => {
    expect(
      getSegmentedButtonColors({
        theme: getTheme(false, false),
        disabled: true,
        checked: false,
      })
    ).toMatchObject({
      textColor: getTheme(false, false).colors.disabled,
    });
  });
});

describe('getDisabledSegmentedButtonBorderWidth', () => {
  it('Returns empty style object for all enabled buttons', () => {
    [0, 1, 2].forEach((index) => {
      expect(
        getDisabledSegmentedButtonStyle({
          theme: getTheme(),
          buttons: [
            { disabled: false },
            { disabled: false },
            { disabled: false },
          ],
          index,
        })
      ).toMatchObject({});
    });
  });

  it('Returns empty style object for all disabled buttons', () => {
    [0, 1, 2].forEach((index) => {
      expect(
        getDisabledSegmentedButtonStyle({
          theme: getTheme(),
          buttons: [{ disabled: true }, { disabled: true }, { disabled: true }],
          index,
        })
      ).toMatchObject({});
    });
  });

  it('Returns proper style object for one disabled button', () => {
    expect(
      getDisabledSegmentedButtonStyle({
        theme: getTheme(),
        buttons: [{ disabled: false }, { disabled: true }, { disabled: true }],
        index: 0,
      })
    ).toMatchObject({ borderRightWidth: 1 });
  });

  it('Returns proper style object for two disabled buttons (alternately)', () => {
    [0, 2].forEach((index) => {
      expect(
        getDisabledSegmentedButtonStyle({
          theme: getTheme(),
          buttons: [
            { disabled: false },
            { disabled: true },
            { disabled: false },
            { disabled: true },
          ],
          index,
        })
      ).toMatchObject({ borderRightWidth: 1 });
    });
  });
});

describe('should have `accessibilityState={ checked: true }` when selected', () => {
  it('should have two button selected', () => {
    const onValueChange = jest.fn();
    const { getAllByA11yState } = render(
      <SegmentedButtons
        multiSelect
        value={['walk', 'transit']}
        buttons={[
          { value: 'walk', label: 'Walking' },
          { value: 'transit', label: 'Transit' },
          { value: 'drive', label: 'Driving' },
        ]}
        onValueChange={onValueChange}
      />
    );

    const checkedButtons = getAllByA11yState({ checked: true });
    expect(checkedButtons).toHaveLength(2);
  });

  it('show selected check icon should be shown', () => {
    const onValueChange = jest.fn();

    const { getByTestId } = render(
      <SegmentedButtons
        multiSelect
        value={['walk', 'transit']}
        buttons={[
          {
            value: 'walk',
            label: 'Walking',
            showSelectedCheck: true,
            testID: 'walking-check-icon',
          },
          { value: 'transit', label: 'Transit' },
          { value: 'drive', label: 'Driving' },
        ]}
        onValueChange={onValueChange}
      />
    );

    expect(getByTestId('walking-check-icon')).toBeDefined();
  });
});
