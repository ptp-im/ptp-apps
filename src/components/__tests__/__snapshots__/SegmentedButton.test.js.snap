// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders checked segmented button with selected check 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(232, 222, 248, 1)",
          "borderBottomRightRadius": 0,
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderRightWidth": 0,
          "borderTopRightRadius": 0,
          "borderWidth": 1,
        },
        Object {
          "borderStyle": "solid",
          "minWidth": 76,
        },
        Array [
          undefined,
          Object {
            "borderRightWidth": 1,
          },
        ],
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "checked": true,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          false,
          Object {
            "borderBottomRightRadius": 0,
            "borderRadius": 20,
            "borderRightWidth": 0,
            "borderTopRightRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 9,
            },
            Object {
              "paddingVertical": 9,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "marginRight": 3,
              "transform": Array [
                Object {
                  "scale": 0,
                },
              ],
            }
          }
          testID="undefined-check-icon"
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontSize": 18,
                },
              ]
            }
          >
            â–¡
          </Text>
        </View>
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
              ],
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 0,
          "borderColor": "rgba(28, 27, 31, 0.12)",
          "borderRadius": 20,
          "borderTopLeftRadius": 0,
          "borderWidth": 1,
        },
        Object {
          "borderStyle": "solid",
          "minWidth": 76,
        },
        Array [
          undefined,
          Object {},
        ],
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "checked": false,
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          false,
          Object {
            "borderBottomLeftRadius": 0,
            "borderRadius": 20,
            "borderTopLeftRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 9,
            },
            Object {
              "paddingVertical": 9,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "marginRight": 0,
            }
          }
        />
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "color": "rgba(28, 27, 31, 0.38)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
              ],
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders disabled segmented button 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(232, 222, 248, 1)",
          "borderBottomRightRadius": 0,
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderRightWidth": 0,
          "borderTopRightRadius": 0,
          "borderWidth": 1,
        },
        Object {
          "borderStyle": "solid",
          "minWidth": 76,
        },
        Array [
          undefined,
          Object {
            "borderRightWidth": 1,
          },
        ],
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "checked": true,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          false,
          Object {
            "borderBottomRightRadius": 0,
            "borderRadius": 20,
            "borderRightWidth": 0,
            "borderTopRightRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 9,
            },
            Object {
              "paddingVertical": 9,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "marginRight": 0,
            }
          }
        />
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
              ],
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 0,
          "borderColor": "rgba(28, 27, 31, 0.12)",
          "borderRadius": 20,
          "borderTopLeftRadius": 0,
          "borderWidth": 1,
        },
        Object {
          "borderStyle": "solid",
          "minWidth": 76,
        },
        Array [
          undefined,
          Object {},
        ],
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "checked": false,
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          false,
          Object {
            "borderBottomLeftRadius": 0,
            "borderRadius": 20,
            "borderTopLeftRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 9,
            },
            Object {
              "paddingVertical": 9,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "marginRight": 0,
            }
          }
        />
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "color": "rgba(28, 27, 31, 0.38)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
              ],
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders segmented button 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(232, 222, 248, 1)",
          "borderBottomRightRadius": 0,
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderRightWidth": 0,
          "borderTopRightRadius": 0,
          "borderWidth": 1,
        },
        Object {
          "borderStyle": "solid",
          "minWidth": 76,
        },
        Array [
          undefined,
          Object {},
        ],
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "checked": true,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          false,
          Object {
            "borderBottomRightRadius": 0,
            "borderRadius": 20,
            "borderRightWidth": 0,
            "borderTopRightRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 9,
            },
            Object {
              "paddingVertical": 9,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "marginRight": 0,
            }
          }
        />
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
              ],
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 0,
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderTopLeftRadius": 0,
          "borderWidth": 1,
        },
        Object {
          "borderStyle": "solid",
          "minWidth": 76,
        },
        Array [
          undefined,
          Object {},
        ],
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "checked": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          false,
          Object {
            "borderBottomLeftRadius": 0,
            "borderRadius": 20,
            "borderTopLeftRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 9,
            },
            Object {
              "paddingVertical": 9,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "marginRight": 0,
            }
          }
        />
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
              ],
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
